misk-admin/src/main/kotlin/misk/web/metadata/all/AllMetadataAction.kt

```diff
@@ -11,15 +12,24 @@ import misk.web.metadata.Metadata
 
 @Singleton
 class AllMetadataAction @Inject constructor(
-  private val metadata: Map<String, Metadata>
+  private val allMetadata: Map<String, Metadata>
 ) : WebAction {
-  @Get("/api/all/metadata")
+  @Get(PATH)
   @RequestContentType(MediaTypes.APPLICATION_JSON)
   @ResponseContentType(MediaTypes.APPLICATION_JSON)
   @AllMetadataAccess
-  fun getAll(): Response {
-    return Response(all = metadata)
+  fun getMetadata(@PathParam id: String): Response {
+    // Return all metadata if the id is "all"
+    if (id == "all") return Response(metadata = allMetadata)
+
+    // Return metadata for the requested id
+    val metadata = allMetadata.filter { it.key == id }
+    return Response(metadata = metadata)
   }
 
-  data class Response(val all: Map<String, Metadata>)
+  data class Response(val metadata: Map<String, Metadata>)
+
+  companion object {
+    const val PATH = "/api/{id}/metadata"
```

Comment:
Now takes a path param, `all` to get all metadata, or the metadata id to get a specific one (ie. `config`).